graphs/frequent_pattern_graph_miner.py

```diff
@@ -0,0 +1,189 @@
+    for i in cluster:
+        support.append(i*100/len(cluster)) 
+        
+    return sorted_frequency_table,cluster,nodes,support   
+
+def print_all():
+    print("\nNodes\n")
+    for key,value in nodes.items():
+        print(key,value)
+    print("\nSupport\n")
+    print(support)
+    '''print("\n Edge List\n")
+    for i in EL:
+        print(i)'''
+    print("\n Cluster \n")
+    for key,value in sorted(cluster.items(),reverse=True):
+        print(key, value)
+    print("\n Graph\n")
+    for key,value in graph.items():
+        print(key, value)
+    print("\n Edge List of Frequent subgraphs \n")
+    for edge_list in freq_subgraph_edge_list:
+        print(edge_list)
+       
+def create_edge(nodes,graph,cluster,c1):
+    '''
+    create edge between the nodes 
+    '''
+    for i in cluster[c1].keys():
+        count=0
+        c2=c1+1
+        while c2 < max(cluster.keys()):
+            for j in cluster[c2].keys():
+                '''
+                creates edge only if the condition satisfies
+                '''
+                if(int(i,2) & int(j,2) == int(i,2)): 
+                   if tuple(nodes[i]) in graph:
+                       graph[tuple(nodes[i])].append(nodes[j])
+                   else:
+                        graph[tuple(nodes[i])]=[nodes[j]]
+                   count+=1
+            if(count==0):
+                c2=c2+1
+            else:
+                break
+
+def construct_graph(cluster,nodes): 
+    X=cluster[max(cluster.keys())]
+    cluster[max(cluster.keys())+1]='Header'
+    graph={}
+    for i in X.keys():
+        if tuple(['Header']) in graph:
+            graph[tuple(['Header'])].append(X[i])
+        else:
+            graph[tuple(['Header'])]=[X[i]]
+    for i in X.keys():
+        graph[tuple(X[i])]=[['Header']]
+    i=1
+    while i < max(cluster.keys())-1:
+        create_edge(nodes,graph,cluster,i) 
+        i=i+1
+        
+    return graph
+
+def myDFS(graph,start,end,path=[]): 
+    '''
+    find different DFS walk from given node to Header node
+    '''
+    path=path+[start] 
+    if start==end:
+        paths.append(path) 
+    for node in graph[start]:
+        if tuple(node) not in path:
+            myDFS(graph,tuple(node),end,path)
+            
+def find_freq_subgraph_given_support(s,cluster,graph):
+    '''
+    find edges of multiple frequent subgraphs
+    '''
+    k=int(s/100*(len(cluster)-1))
+    freq_subgraphs=[]
+    for i in cluster[k].keys():
+        myDFS(graph,tuple(cluster[k][i]),tuple(['Header']))
+
+def freq_subgraphs_edge_list(paths):
+    '''
+    returns Edge list for frequent subgraphs
+    '''
+    freq_sub_EL=[]
+    for edges in paths:
+        EL=[]
+        for j in range(len(edges)-1):
+            temp=list(edges[j])
+            for e in temp:
+                edge=(e[0],e[1])
+                EL.append(edge)
+        freq_sub_EL.append(EL)
+        
+    return freq_sub_EL 
+def preprocess(edge_array: List[List[str]) -> List[List[List[str]]]:

```

Comment:
```suggestion
def preprocess(edge_array: List[List[str]]) -> List[List[List[str]]]:

```

Reply:
Also, here we are making the promise that this function with return `List[List[List[str]]]` but the function has no return statement so it only returns `None`.